open-apparel-registry/open-apparel-registry
# Add Craco and Prettier

## Overview

Move to the standard Craco/Prettier configuration we have in other projects, allowing for partially automated code formatting. Upgrade to React-Scripts 4.0.2 and React 16. 

Connects #1266, #1252  

## Notes

Fast-reloading is enabled by default in CRA 4, but while it is correctly compiling the updated code it is not auto-refreshing. This [might be due to Vagrant settings](https://create-react-app.dev/docs/troubleshooting/#npm-start-doesnt-detect-changes). Turning off the fast reloading allows for the hot-module reloading, which restores the previous behavior of auto-refreshes in the React app. We should continue to look into how to enable the fast refresh within the Vagrant environment. 

## Testing Instructions

* On this branch, run `vagrant ssh` and `vagrant update`
* Run `./scripts/lint` and confirm no necessary changes are found
* Run `./scripts/server` and confirm that the website behavior hasn't changed
* Make a change in a file and save. The app should reload as before. 

## Checklist
